CONTRIBUTING.md

```diff
@@ -0,0 +1,313 @@
+
+Testing of MySQL is a whole big thing, so please go to
+the [ official documentation on MySQL testing ](https://dev.mysql.com/doc/dev/mysql-server/latest/PAGE_TESTING_TOOLS.html)
+first.
+
+If it is bug fix, make sure that corresponding tests from the test suite work fine. It's a good idea to patch existing
+tests to cover your specific case. For the introduction of new features, the development of a comprehensive test suite is strongly recommended.
+
+## Preparing the Pull Request
+
+After finishing the tasks outlined above, you must prepare your branch for a pull request.   
+Please do not forget to run ``` git clang-format ``` on your code before pushing to the GitHub repository.
+
+Then, squash all commits to one or a couple of logically consistent commits.
+For example, the fix of bug should have just one commit. If you have a new feature that requires some changes to the
+existing code, then the first commit should contain required changes, and the second is a new feature and tests.
+Commit message should contain the Jira issue number and then a short but explanatory description in the same line. Th next line is usually the URL of the Jira issue, and then goes the full description of the commit. The commit message is very important for us, because we do a lot of work with Git wile merging upstream releases and commit messages help us to resolve possible conflicts.
+Example of the commit message:
+```
+PS-9876 Fix of the compression bug with zlib call parameters
+
+https://perconadev.atlassian.net/browse/PS-9876
+
+This code fixes compression function call parameters by proper calculation of the
+offset in case of previous error.
+```
+ 
+It is not a dogma, on the other hand, we value a clean and consistent Git history. Please help us maintain this standard.
+
+Now you can push your local branch to your GitHub repository and make the pull request to the Percona repository.
+
+Please, write meaningful title for the PR and put the number of Jira issue at the beginning of the first line in the
+description. The second line is usually a link to the Jira issue. Then the short description follows. No need to write a
+long description because you wrote one in the Jira issue.
+
+Example of a pull request form:
+
+Title:
+```
+PS-9876 Fix of the compression bug
+```
+Description:
+```
+PS-9876 Fix of the compression bug
+
+https://perconadev.atlassian.net/browse/PS-9876
+
+The first commit is update of zlib, the second is the fixes of compression function calls.
+```
+
+### Peer review and requests of changes
+
+Peer review of the code always goes before PR merge. So, you have to add some core development team members in the "Reviewers" field of PR. Assign the PR to Yura Sorokin, our team lead, and then he will add other developers if needed. 
```

Comment:
Noooooooooooooo!
Do not assign  this to "Yura Sorokin" 🙃

Reply:
Yeah... Life sucks :)